### ISSUE TITLE
Walking into Projectile Doesn't Activate it

### ISSUE THREAD

**Issue by Snayff** [Snayff/notquiteparadise#228]
## The Bug
When an entity walks into a projectile the projectile doesn't activate. Then the projectile's next turn it tries and fails to move and resolves it's collision effect. The projectile also applies the `MoveEffect` to the blocking entity.  

## Reproduction Instructions
1. Entity 1 casts a projectile at another. 
2. Entity 2, a valid target of the projectile cast by Entity 1, walks into the projectile. 

## Expected behaviour
Moving into the projectile should cause it to activate if the moving entity is a valid target. 

## Platform
Describe the system and setup you were using when the bug happened. 

 - OS: Windows 10
 - Python Version: 3.8
 - NQP Version: 0.141.1

## Log Output/Stack Trace
```
2020-12-31 15:59:50,621| DEBUG   | 'player' used 'Move'.
2020-12-31 15:59:50,622| DEBUG   | Evaluating Move Actor Effect...
2020-12-31 15:59:50,622| INFO    | 'player' paid 0 stamina and has 500 left.
2020-12-31 15:59:50,622| INFO    | Moving to the next turn...
2020-12-31 15:59:50,622| DEBUG   | Building a new turn queue...
2020-12-31 15:59:50,623| DEBUG   | -> New queue built: [('player', 482), ('Krock', 493), ('Krocks Splash`s projectile', 462)]
2020-12-31 15:59:50,623| DEBUG   | -> Current time is 462. We are 1 TU`s into round 24.
2020-12-31 15:59:50,623| DEBUG   | -> 'Krocks Splash`s projectile' is now the turn holder.
2020-12-31 15:59:50,626| INFO    | Moving to the next turn...
2020-12-31 15:59:50,626| DEBUG   | Building a new turn queue...
*removed queue*
2020-12-31 15:59:50,638| DEBUG   | 'Krocks Splash`s projectile' is beginning their turn.
2020-12-31 15:59:50,638| DEBUG   | 'Krocks Splash`s projectile' has 2 range left and is going to move from (9,12) to (8,12).
2020-12-31 15:59:50,638| DEBUG   | 'Krocks Splash`s projectile' used 'Move'.
2020-12-31 15:59:50,639| DEBUG   | Evaluating Move Actor Effect...
2020-12-31 15:59:50,639| DEBUG   | Evaluating Move Actor Effect...
2020-12-31 15:59:50,639| DEBUG   | 'Krocks Splash`s projectile' tried to move in LEFT to (8,12) but was blocked by 'player'. 
2020-12-31 15:59:50,640| INFO    | Moving to the next turn...
2020-12-31 15:59:50,640| DEBUG   | Building a new turn queue...
*removed queue*
2020-12-31 15:59:50,643| INFO    | Moving to the next turn...
2020-12-31 15:59:50,643| DEBUG   | Building a new turn queue...
*removed queue*
2020-12-31 15:59:50,654| DEBUG   | 'Krocks Splash`s projectile' is beginning their turn.
2020-12-31 15:59:50,655| DEBUG   | 'Krocks Splash`s projectile' is going to activate at (9,12).
2020-12-31 15:59:50,655| DEBUG   | Evaluating Apply Affliction Effect...
*removed evaluation*
2020-12-31 15:59:50,655| DEBUG   | Evaluating Damage Effect...
*removed evaluation*
2020-12-31 15:59:50,656| INFO    | 'player' takes 9 and has 143 health remaining.
2020-12-31 15:59:50,657| INFO    | 'Krocks Splash`s projectile' (15) added to stack to be deleted on next frame.
```

**Comment by DaFluffyPotato** [comment]
It looks like this is only an issue when moving diagonally. I moved straight into the projectile in these 2 images and I took damage.
![img_1](https://user-images.githubusercontent.com/28032741/105375584-d46d6480-5bd6-11eb-9a24-e588a371148e.png)
![img_0](https://user-images.githubusercontent.com/28032741/105375601-d800eb80-5bd6-11eb-8a80-c2a93d193d6d.png)
Is there supposed to be knockback from the projectile? I appear to have been knocked back when I took that hit.

Anyways, I didn't look into what you did to "resolve" the diagonal movement issue (#122). Did you decide diagonal movement was okay? And if not, did you fix the double move skill cast from it? Either way, this appears to be a side effect of whatever is going on with the movement. You should probably open another issue (or update this one) with the intentions for movement.

**Comment by Snayff** [comment]
Diagonal movement shouldn't be possible for the Move skill. #122 was, as you identified, an issue of 2 events being generated before the state changed and that should be resolved now. 

No knockback intended, that seems to be the projectile affecting both itself and the hit entity with the move effect, instead of only affecting itself.

**Comment by DaFluffyPotato** [comment]
Should I reopen #122 again and take care of it then? I can also open an issue for whatever happened with that "knockback" and assign it to myself.

**Comment by Snayff** [comment]
Please do.

**Comment by DaFluffyPotato** [comment]
manifestation of #122, so it's a duplicate
